### ISSUE TITLE
Add callback and widget_template parameters for awful.widget.layoutlist

### ISSUE THREAD

**Issue by erik-f** [awesomeWM/awesome#2950]
In widgets like awful.widget.tasklist and awful.widget.taglist one can pass a callback function and a widget template, so it's easy to change the whole widget.

In awful.widget.layoutlist it's only possible to set the screen and buttons, so it's very inconvenient to change the behaviour of this widget.

**Comment by Elv13** [comment]
Can you provide an API example of what you would like to see?

**Comment by erik-f** [comment]
Of course. So right now the constructor of awful.widget.layoutbox has the parameters `args.screen` and `args.buttons`.
The layoutbox widget looks like this (copied from the source code):

    wibox.widget {
        {
            id     = "imagebox",
            widget = wibox.widget.imagebox
        },
        {
            id     = "textbox",
            widget = wibox.widget.textbox
        },
        layout = wibox.layout.fixed.horizontal
    }

and the function to update looks like this:

    local function update(w, screen)
        screen = get_screen(screen)
        local name = layout.getname(layout.get(screen))
        w._layoutbox_tooltip:set_text(name or "[no name]")

        local img = surface.load_silently(beautiful["layout_" .. name], false)
        w.imagebox.image = img
        w.textbox.text   = img and "" or name
    end

I'd like to be able to pass a custom template to build the widget, so another parameter `args.widget_template` in the constructor with the default value being the first code I copied here.
Also I'd like to pass a custom callback function to update the widget, so another parameter `args.callback` with the default value being the function above.

**Comment by SethBarberee** [comment]
Taking a crack at this PR. Hoping to have something in the coming week.
